samples/query-features-with-arcade-expression/src/main/java/com/esri/arcgismaps/sample/queryfeatureswitharcadeexpression/components/QueryFeaturesWithArcadeExpressionViewModel.kt

```diff
@@ -0,0 +1,187 @@
+import com.arcgismaps.data.ArcGISFeature
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.PortalItem
+import com.arcgismaps.mapping.layers.Layer
+import com.arcgismaps.mapping.symbology.PictureMarkerSymbol
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.ScreenCoordinate
+import com.arcgismaps.portal.Portal
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.queryfeatureswitharcadeexpression.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+class QueryFeaturesWithArcadeExpressionViewModel(application: Application) :
+    AndroidViewModel(application) {
+
+    // setup the red pin marker image as a bitmap drawable
+    private val markerDrawable: BitmapDrawable by lazy {
+        val bitmap = BitmapFactory.decodeResource(application.resources, R.drawable.map_pin_symbol)
+        BitmapDrawable(application.resources, bitmap)
+    }
+
+    // setup the red pin marker as a Graphic
+    private val markerGraphic: Graphic by lazy {
+        val markerSymbol = PictureMarkerSymbol.createWithImage(markerDrawable).apply {
+            width = 30f
+            height = 30f
+            offsetY = 25f
+        }
+
+        Graphic(symbol = markerSymbol)
+    }
+
+    // data layer to be loaded from portal item
+    private var policeBeatsLayer: Layer? = null
+
+    // state flow to expose query results and status to UI
+    private val _queryStateFlow = MutableStateFlow(QueryState())
+    val queryStateFlow = _queryStateFlow.asStateFlow()
+
+    val graphicsOverlay = GraphicsOverlay()
+
+    // create a portal item with the itemId of the web map
+    val portal = Portal("https://www.arcgis.com/")
+    val portalItem = PortalItem(portal, "539d93de54c7422f88f69bfac2aebf7d")
+
+    // create and add a map with with portal item
+    val arcGISMap = ArcGISMap(portalItem)
+
+    // create a map view proxy for handling interactions with the map view
+    val mapViewProxy = MapViewProxy()
+
+    // create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    init {
+        viewModelScope.launch {
+            arcGISMap.load().onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    "Failed to load map",
+                    error.message.toString()
+                )
+            }
+        }
+
+        // add the marker graphic to the graphics overlay
+        graphicsOverlay.graphics.add(markerGraphic)
+
+        // get the RPD Beats layer from the map's operational layers
+        policeBeatsLayer = arcGISMap.operationalLayers.firstOrNull { layer ->
+            layer.id == "RPD_Reorg_9254"
+        }
```

Comment:
I think this needs to go in the coroutine scope above, because if the map is not loaded then the layer remains null